### ISSUE TITLE
Publish and distribute AtlasMapperUI as a compiled JavaScript artifact from npm

### ISSUE THREAD

**Issue by deeleman** [syndesisio/syndesis#424]
### TL;DR
This update is required in order to upgrade Syndesis UI to Angular 5, since AtlasMapper will be removed from the compilation output by the Angular CLI.

### Detailed spec
We are currently distributing AtlasMapper UI as a TypeScript module through npmjs.org. Ideally, no package distributed from npmjs is meant to be built from the client application. This acquires special relevance on TypeScript applications, where the `tsconfig.json` setup usually excludes the `node_modules` folder. On the other hand, the typescript version that AtlasMapper uses isn't necessarily the same that the one its client (eg. Syndesis UI) might be using, leading to compilation errors or unexpected behaviors.

The purpose of this issue is to add a binary distribution to the repo so we can distribute a safe version of AtlasMapper from npmjs.org.

**Comment by deeleman** [comment]
All right, so I wanted to share a status of this issue and its impact on other related issues:

1. The Syndesis upgrade to Angular 5 is blocked, because AtlasMapper is fetched from npmjs.org in raw TypeScript format, forcing Syndesis to compile it before compiling itself. Because of the reasons mentioned before we cannot do this anymore.
2. In order to turn AtlasMapper into a component library I need to update a good bunch of its dependencies, which have remain unaltered for 5 months and are now quite outdated.
3. This also requires us to tweak the folder structure quite a bit to ensure that we can still leverage the dev commands in package.json plus bootstrap the demo app

This will turn into a long PR I'm afraid.

**Comment by rhuss** [comment]
@deeleman is this issues still relevant (and is i.e. still a blocker) ? If so, it is a blocker for TP3, too ? (and we have to move it to this sprint).

**Comment by deeleman** [comment]
Done
